File: Linked List/removeKthElementFromLast.js
class  ListNode {
    constructor(val){
      this.val = val;
      this.next = null;
  }
}
class LL{
    constructor(){
        this.root = new ListNode(-1);

    }
    insert(data){
        let current = this.root;
        while(current.next != null){
            current = current.next;
        }
        const end = new ListNode(data);
        current.next = end;
    }
    display(){
        let current = this.root.next;
        while(current != null){
            console.log(current.val);
            current = current.next;
        }
    }
    removeNthFromEnd(n){
        let head = this.root.next;
        while(true){
            let current = head;
            for(let i = 0;i<n;i++){
                current = current.next;
            }
            if(current.next == null){
                head.next = head.next.next;
                break;
            }
            head = head.next;
        }

    }
}

const ll = new LL();
for(let i of [1]){
    ll.insert(i)
}

ll.display()
console.log("\n");
ll.removeNthFromEnd(1);
ll.display()

File: Linked List/removeKthElementFromLast.py
class Node():
    def __init__(self, data, next=None):
        self.data = data
        self.next = next


class LL():
    def __init__(self):
        self.root = Node(-1)

    def insert(self, data):
        current = self.root
        while current.next != None:
            current = current.next
        end = Node(data)
        current.next = end

    def display(self):
        current = self.root.next
        while current:
            print(current.data)
            current = current.next

    def removeKthFromLast(self, n):
        dummy = Node(0, self.root.next)
        p1 = dummy
        p2 = self.root.next
        for _ in range(n):
            if p2 == None:
                break
            p2 = p2.next
        while p2:
            p1 = p1.next
            p2 = p2.next
        p1.next = p1.next.next


ll = LL()

for i in [1]:
    ll.insert(i)
ll.display()
ll.removeKthFromLast(1)
print("\n")
ll.display()
